src/tcp/_Private/DisposableTCPSocket.php

```diff
@@ -0,0 +1,28 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace HH\Lib\_Private;
+
+use namespace HH\Lib\Experimental\{IO, TCP};
+
+final class DisposableTCPSocket
+  extends DisposableHandleWrapper<TCP\NonDisposableSocket>
+  implements
+    \IAsyncDisposable,
+    IO\DisposableReadWriteHandle,
+    TCP\DisposableSocket {
+
+  use DisposableReadHandleWrapperTrait<TCP\NonDisposableSocket>;
+  use DisposableWriteHandleWrapperTrait<TCP\NonDisposableSocket>;
+
+  public function __construct(TCP\NonDisposableSocket $impl) {
+    parent::__construct($impl);
+  }
```

Comment:
This just makes the constructor public, right? Why not make the constructor public on `DisposableHandleWrapper` so that we don't need this in every subclass?

Reply:
That'll force all concrete implementations to be in the _Private NS; subclasses can not have narrower constructor visibility :/

That's probably actually fine, I'm just being lazy :)